tests/utils_tests/test_http.py

```diff
@@ -317,8 +318,22 @@ def test_parsing_rfc1123(self):
         self.assertEqual(datetime.utcfromtimestamp(parsed), datetime(1994, 11, 6, 8, 49, 37))
 
     def test_parsing_rfc850(self):
-        parsed = parse_http_date('Sunday, 06-Nov-94 08:49:37 GMT')
-        self.assertEqual(datetime.utcfromtimestamp(parsed), datetime(1994, 11, 6, 8, 49, 37))
+        # fix current date for testing rollover >50 years in future
+        mocked_datetime = Mock(
+            utcnow=Mock(return_value=datetime(2019, 11, 6, 8, 49, 37)),
+            side_effect=lambda *args, **kw: datetime(*args, **kw)
+        )
```

Comment:
I would move mocking `datetime` to a decorator, after that we will be able to test different dates, e.g.
```python
@mock.patch('django.utils.http.datetime.datetime')
def test_parsing_rfc850(self, mocked_datetime):
    mocked_datetime.side_effect = lambda *args, **kw: datetime(*args, **kw)
    utcnow_first_fifty = datetime(2019, 11, 6, 8, 49, 37)
    utcnow_second_fifty = datetime(2051, 11, 6, 8, 49, 37)
    date = (
        ('Tuesday, 31-Dec-69 08:49:37 GMT', datetime(2069, 12, 31, 8, 49, 37), utcnow_first_fifty),
        ('Monday, 10-Nov-70 18:49:37 GMT', datetime(1970, 11, 10, 18, 49, 37), utcnow_first_fifty),
        ('Wednesday, 31-Dec-71 18:49:37 GMT', datetime(1971, 12, 31, 18, 49, 37), utcnow_first_fifty),
        ('Thursday, 31-Dec-99 08:49:37 GMT', datetime(2099, 12, 31, 8, 49, 37), utcnow_second_fifty),
        ('Thursday, 10-Nov-50 18:49:37 GMT', datetime(2050, 11, 10, 18, 49, 37), utcnow_second_fifty),
        ('Sunday, 31-Dec-00 18:49:37 GMT', datetime(2000, 12, 31, 18, 49, 37), utcnow_second_fifty),
    )
    for rfc850str, expected_date, utcnow in date:
        mocked_datetime.utcnow = mock.Mock(return_value=utcnow)
        with self.subTest(string=rfc850str):
            parsed = parse_http_date(rfc850str)
            self.assertEqual(datetime.utcfromtimestamp(parsed), expected_date)
```

Reply:
Thank you, great idea.